### PR TITLE
add the masters countdown

### PR DESCRIPTION
**Pull request by imagesafari**

### CODE REVIEW

**Review by mjakefranklin — APPROVED** [review]

File: scripts/qrzgrid.tcl [hunk 1]
```diff
@@ -2129,5 +2129,21 @@ proc msg_spot {nick uhand handle input} {
 	close $fd
 }
 
+bind pub - !masters masters
+bind pub - !themasters masters
+set mastersbin "/home/eggdrop/bin/masters"
+proc fieldday { nick host hand chan text } {
```
Comment (molo1134) [inline scripts/qrzgrid.tcl:2135]: need to change this proc name, or else it will conflict with the `!fieldday` command

File: lib/masters [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+use Date::Manip::Date;
+
+use File::Basename;
+use Cwd 'realpath';
+use lib dirname(realpath(__FILE__));
+use Util;
+
+my $d1 = DateTime->now;
+my ($curryear, $currmon, $currdate) = split / /, strftime("%Y %m %d", gmtime($d1->epoch()));
+my $fddate = new Date::Manip::Date;
```
Comment (molo1134) [inline lib/masters:22]: should probably change variablenames from `$fd____` to `$m____` as needed.

File: lib/masters [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+$fddate->parse('Second Sunday in April');
+my ($fdyear, $fdmon, $fdday) = split / /, $fddate->printf('%Y %m %d');
+
+my $d2 = DateTime->new(
+    year       => $fdyear,
+    month      => $fdmon,
+    day        => $fdday - 3,
+    hour       => 12, # 1800 UTC
```
Comment (molo1134) [inline lib/masters:30]: confused by this, is it supposed to be 1800 UTC or 1200 UTC?

File: lib/masters [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    minute     => 0,
+    second     => 0,
+);
+
+my $d3 = DateTime->new(
+    year       => $fdyear,
+    month      => $fdmon,
+    day        => $fdday,
+    hour       => 23, # 1800 UTC
```
Comment (molo1134) [inline lib/masters:39]: same here

File: scripts/qrzgrid.tcl
Comment (molo1134) [inline scripts/qrzgrid.tcl:1]: this should go into fun.tcl instead of qrzgrid.tcl

**Review by molo1134 — CHANGES_REQUESTED** [review]
functionality works, but some minor changes needed.

```
$ faketime '2024-04-10' ./masters 
The Masters Countdown! 1 day, 8 hours, 0 min, 0 sec
$ faketime '2024-04-11' ./masters
The Masters Countdown! 8 hours, 0 min, 0 sec
$ faketime '2024-04-12' ./masters
Hello, friends. A tradition unlike any other. https://masters.com
$ faketime '2024-04-13' ./masters
Hello, friends. A tradition unlike any other. https://masters.com
$ faketime '2024-04-14' ./masters
Hello, friends. A tradition unlike any other. https://masters.com
$ faketime '2024-04-15' ./masters
The Masters Countdown! 12 months, 1 day, 7 hours, 59 min, 59 sec
$ faketime '2024-04-16' ./masters
The Masters Countdown! 12 months, 8 hours, 0 min, 0 sec
```

**Comment by molo1134** [comment]
:+1: TU

**Pull request merged**